packages/api-plugin-bull-queue/src/api/createQueue.js

```diff
@@ -0,0 +1,57 @@
+import { createRequire } from "module";
+import Queue from "bull";
+import ms from "ms";
+import Logger from "@reactioncommerce/logger";
+import config from "../config.js";
+
+const require = createRequire(import.meta.url);
+
+const pkg = require("../../package.json");
+
+const { name, version } = pkg;
+const logCtx = {
+  name,
+  version,
+  file: "api/createQueue.js"
+};
+
+
+const {
+  JOBS_SERVER_REMOVE_COMPLETED_JOBS_AFTER,
+  JOBS_SERVER_REMOVE_FAILED_JOBS_AFTER,
+  REACTION_WORKERS_ENABLED
+} = config;
+
+
+const defaultOptions = {
+  removeOnComplete: { age: ms(JOBS_SERVER_REMOVE_COMPLETED_JOBS_AFTER) },
+  removeOnFail: { age: ms(JOBS_SERVER_REMOVE_FAILED_JOBS_AFTER) }
+};
+
+const { REDIS_SERVER } = config;
+
+/**
+ * @summary create a new named instance of the BullMQ Queue
+ * @param {Object} context - The application context
+ * @param {String} queueName - The name of the queue to create, this name is used elsewhere to reference the queue
+ * @param {Object} options - Any additional options to pass to the instance
+ * @param {Function} processorFn - The processor function to use for jobs in the queue
+ * @return {Object|Boolean} - An instance of a BullMQ queue
+ */
+export default function createQueue(context, queueName, options = defaultOptions, processorFn) {
+  if (typeof queueName !== "string" || typeof options !== "object" || typeof processorFn !== "function") {
```

Comment:
Should the `typeof processorFn !== "function"` check be dependent on whether `REACTION_WORKERS_ENABLED`? I could see arguments for requiring it always, but it truthfully wouldn't be needed if this instance won't run workers.